lib/puma/dsl.rb

```diff
@@ -581,6 +581,11 @@ def threads(min, max = min)
         raise "The maximum number of threads (#{max}) must be greater than 0"
       end
 
+      if max == 1
+        warn "[WARNING] Puma config `threads #{min}, #{max}` has a max thread count of 1. If the application makes"
+        warn "          blocking HTTP calls to itself, it may deadlock. Consider increasing the max thread count."
+      end
```

Comment:
This is a questionable warning to add, isn't running multiple workers each with one thread a legitime use-case with Puma?

```shell
$ echo 'app { [200, {}, ["OK"]] }' | ruby -Ilib ./bin/puma --config /dev/stdin --port 0 --log-requests --workers 3 --threads 1:1
[WARNING] Puma config `threads 1, 1` has a max thread count of 1. If the application makes
          blocking HTTP calls to itself, it may deadlock. Consider increasing the max thread count.
[70321] Puma starting in cluster mode...
[70321] * Puma version: 6.4.3 (ruby 3.3.5-p100) ("The Eagle of Durango")
[70321] *  Min threads: 1
[70321] *  Max threads: 1
[70321] *  Environment: development
[70321] *   Master PID: 70321
[70321] *      Workers: 3
[70321] *     Restarts: (✔) hot (✔) phased
[70321] * Listening on http://0.0.0.0:64174
[70321] Use Ctrl-C to stop
[70322] + Gemfile in context: /Users/dentarg/src/puma/Gemfile
[70323] + Gemfile in context: /Users/dentarg/src/puma/Gemfile
[70324] + Gemfile in context: /Users/dentarg/src/puma/Gemfile
[70321] - Worker 0 (PID: 70322) booted in 0.0s, phase: 0
[70321] - Worker 1 (PID: 70323) booted in 0.0s, phase: 0
[70321] - Worker 2 (PID: 70324) booted in 0.0s, phase: 0
^C[70321] - Gracefully shutting down workers...
[70321] === puma shutdown: 2024-11-07 21:18:42 +0100 ===
[70321] - Goodbye!
```